internal/rows/arrowbased/arrowRows.go

```diff
@@ -688,3 +670,54 @@ func (vcm *arrowValueContainerMaker) makeColumnValueContainer(t arrow.DataType,
 		return nil, errors.Errorf(errArrowRowsUnhandledArrowType(t.String()))
 	}
 }
+
+// Container for a set of arrow records
+type sparkArrowBatch struct {
+	// Delimiter indicating the range of rows covered by the arrow records
+	rowscanner.Delimiter
+	arrowRecords []SparkArrowRecord
+}
+
+var _ SparkArrowBatch = (*sparkArrowBatch)(nil)
+
+func (b *sparkArrowBatch) Next() (SparkArrowRecord, error) {
+	if len(b.arrowRecords) > 0 {
+		r := b.arrowRecords[0]
+		// remove the record from the slice as iteration is only forwards
+		b.arrowRecords = b.arrowRecords[1:]
```

Comment:
just curious, how efficient is this [1:] operation in go?

Reply:
Very fast the new slice is just pointing in the same memory as the original slice.